tron/kubernetes.py

```diff
@@ -148,35 +148,58 @@ def handle_event(self, event: Event) -> None:
                 self.log.error(f"Event with >1 || 0 containers: {raw_object}")
             else:
                 main_container_statuses = container_statuses[0]
-                main_container_state = main_container_statuses.get("state")
-                if main_container_state is None or main_container_state.get("terminated") is None:
+                main_container_state = main_container_statuses.get("state", {}) or {}
+                main_container_last_state = main_container_statuses.get("lastState", {}) or {}
+                if (main_container_state is None or main_container_state.get("terminated") is None) and (
+                    main_container_last_state is None or main_container_last_state.get("terminated") is None
+                ):
                     self.log.error("Got an event with missing state - assuming success.")
                     self.log.error(f"Event with missing state: {raw_object}")
                 else:
```

Comment:
hmm, i wonder if we should extract some of this logic into a function (or functions) - the initial abnormal exit code was already a bit pushing it in terms of inlining code :p

Reply:
I agree, lets add this in a separate PR though :)